### PR TITLE
New-Azvmss OMode

### PR DESCRIPTION
**Pull request by Sandido**
The Dev and Testing work for New-AzVmss OMode. 
Some simplification for readability is possible, and 1 more condition is required. I plan to move the large if-else blocks to a separate Engine class. 

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1 [hunk 1]
```diff
@@ -2680,64 +2699,99 @@ function Test-VirtualMachineScaleSetOrchestrationModeFaultDomain
...
 }
 
-<#
-.SYNOPSIS
-    testing encryptionAtHost cmdlet for
-    new-azvmss - create vmss using simple parameter set and hostencryption tag.
-    update-azvmss test boolean parameter 
-    new-azvmssconfig
-#>
-function Test-VirtualMachineScaleSetAssignedHost
```
Comment (Sandido) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1:2707]: Issue occurred during merge. Will add this back in.

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs [hunk 2]
```diff
@@ -88,32 +94,152 @@ public override void ExecuteCmdlet()
 
         private void CheckOrchestrationModeRequirementsDefaultParameterSet()
         {
-            if (this.VirtualMachineScaleSet.VirtualMachineProfile != null)
+            if (this.VirtualMachineScaleSet != null)
```
Comment (Sandido) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs:97]: This if check was moved to prior to this method call, no longer needed here.

**Comment by msJinLei** [comment]
@Sandido  Can you fix the static analysis issue https://dev.azure.com/azure-sdk/public/_build/results?buildId=517814&view=logs&j=e1b7b984-9f58-529f-7c5a-f15f8e35cfa6&t=e8689ae8-b216-55f7-b207-4831f84ff98a&l=1543

**Comment by isra-fel** [comment]
Kindly ping @Sandido

**Comment by Sandido** [comment]
This PR was abandoned.

**Pull request closed without merging**